src/bootstrap/step.rs

```diff
@@ -1209,11 +1209,17 @@ invalid rule dependency graph detected, was a rule added and maybe typo'd?
             if paths.len() == 0 && rule.default {
                 Some((rule, 0))
             } else {
-                paths.iter().position(|path| path.ends_with(rule.path))
+                paths.iter()
+                     .position(|path| path.ends_with(rule.path))
                      .map(|priority| (rule, priority))
             }
         }).collect();
 
+        if rules.is_empty() {
```

Comment:
Hm, this is good, but I sort of expected we'd go a step further and check that every path matched with a rule... I might be wrong, but I think that should always be the case. cc @alexcrichton -- can you confirm we should match all paths?